internal/util/util.go

```diff
@@ -128,3 +128,14 @@ func ReadLine(r *bufio.Reader) (string, error) {
 	}
 	return string(ln), err
 }
+
+// LocationProvider is an interface that returns the current location. The
+// github.com/ooni/probe-engine/session.Session implements it.
+type LocationProvider interface {
+	ProbeASN() uint
+	ProbeASNString() string
+	ProbeCC() string
+	ProbeIP() string
+	ProbeNetworkName() string
+	ResolverIP() string
+}
```

Comment:
This is the interface of a thing that tells you about its location. (We basically have replaced a structure exported by another repo with a private interface.) Note that, while it may be tempting to say "export the interface from the probe-engine package", that would not be idiomatic Go. Usually structures are used an interfaces are only used for mocking, internally, or when they are really needed (see `net.Conn`).

Reply:
Xref: https://twitter.com/davecheney/status/942593128355192832

Reply:
And https://github.com/golang/go/wiki/CodeReviewComments#interfaces